app/services/tradeset_description_populator_service.rb

```diff
@@ -0,0 +1,92 @@
+require 'tariff_synchronizer/file_service'
+
+class TradesetDescriptionPopulatorService
+  FILENAME_REGEX = /\A(?<month>[a-zA-Z]+)(?<year>\d+).*\z/
+  CHUNK_SIZE = 5000
+  ENCODINGS = [
+    'Windows-1252:UTF-8',
+    'UTF-16LE:BOM|UTF-8',
+    'ISO-8859-1:UTF-8',
+  ].freeze
+
+  def call
+    each_tradeset_descriptions_chunk do |chunk|
+      upsert_chunk(chunk)
+    end
+  end
+
+  private
+
+  def upsert_chunk(chunk)
+    TradesetDescription.dataset.insert_conflict(
+      constraint: :tradeset_descriptions_filename_description_goods_nomenclatu_key,
+      update: { updated_at: Sequel::CURRENT_TIMESTAMP },
+    ).multi_insert(chunk.uniq)
+  end
+
+  def each_tradeset_descriptions_chunk
+    csv = CSV.open(file_path, headers: true, header_converters: :symbol, encoding:)
+    created_at = Time.zone.now.utc
+    updated_at = created_at
+
+    chunk = []
+
+    csv.each do |row|
+      attributes = row.to_hash
+      attributes[:classification_date] = classification_date
+      attributes[:filename] = filename
+      attributes[:created_at] = created_at
+      attributes[:updated_at] = updated_at
+
+      tradeset_description = TradesetDescription.build(attributes)
+
+      unless chunk.size >= CHUNK_SIZE
```

Comment:
we could use if instead of unless, it's shorter ;) 

`if chunk.size < CHUNK_SIZE`